File: Lab_2_1_cpp/SymbolString.cpp
#include "SymbolString.h"

SymbolString::SymbolString() : BaseString() {}

SymbolString::SymbolString(std::string val) : BaseString(val) {}

SymbolString::SymbolString(const SymbolString& other) : BaseString(other) {}

void SymbolString::replaceChar(char oldChar, char newChar)
{
    for (int i = 0; i < value.length(); i++)
    {
        if (value[i] == oldChar)
            value[i] = newChar;
    }
}

File: Lab_2_1_cs/SymbolString.cs
using System;

namespace StringLibrary
{
    public class SymbolString : BaseString
    {
        public SymbolString() : base() { }

        public SymbolString(string val) : base(val) { }

        public SymbolString(SymbolString other) : base(other) { }

        public void ReplaceChar(char oldChar, char newChar)
        {
            value = value.Replace(oldChar, newChar);
        }
    }
}